### ISSUE TITLE
Issue while trying to deserialize List<CustomClass>

### ISSUE THREAD

**Issue by zzorba** [flipkart-incubator/hbase-object-mapper#5]
It seems like if I create a record with a column of type

```
@HBColumn(family = "cf1", column = "embeds")
List<EmbeddedRecord> embeddedRecords
```
It doesn't fail to compile or flag it as a problem during serialization/deserialization.  However when the class is deserialized it gets stored a List<LinkedHashMap> instead of List<EmbeddedRecord>, causing a class cast exception when you access the field.

**Comment by m-manu** [comment]
@zzorba Currently, although there is support for `List<Integer>`, there isn't support for `List<CustomClass>`. I'm planning to build support for generics of custom classes in upcoming version.

You can however, achieve your intended functionality by encapsulating `List<EmbeddedRecord>` into a class (e.g. `EmbeddedRecordList`).

But I understand your point. If this library doesn't handle `List<CustomClass>` yet, it should throw an explicit exception at the time of serialization itself. I'll fix the issue. Thanks for pointing out.

**Comment by m-manu** [comment]
@zzorba This issue is now fixed in [v1.5.1](https://github.com/flipkart-incubator/hbase-object-mapper/releases/tag/v1.5.1). Serialization/Deserialization of `List<EmbeddedRecord> embeddedRecords` will work fine now.
